BlueDBOnDisk/src/main/java/org/bluedb/disk/segment/Segment.java

```diff
@@ -335,15 +327,11 @@ protected RollupTarget toRollupTarget(Range range) {
 	}
 
 	protected List<Range> getRollupRanges(Range currentChunkRange) {
-		long currentChunkSize = currentChunkRange.length();
-		List<Long> rollupLevelsLargerThanChunk = Blutils.filter(rollupLevels, (l) -> l > currentChunkSize);
-		CheckedFunction<Long, Range> toRange = (l) -> Range.forValueAndRangeSize(currentChunkRange.getStart(), l);
-		List<Range> possibleRollupRanges = Blutils.mapIgnoringExceptions(rollupLevelsLargerThanChunk, toRange);
-		List<Range> rollupRangesEnclosingChunk = Blutils.filter(possibleRollupRanges, (r) -> r.encloses(currentChunkRange));
-		if (currentChunkRange.getEnd() < segmentRange.getStart()) {
-			rollupRangesEnclosingChunk.add(preSegmentRange);
-		}
-		return rollupRangesEnclosingChunk;
+		long currentChunkStart = currentChunkRange.getStart();
+		return calculatePossibleChunkRanges(currentChunkStart).stream()
+				.filter((r) -> r.encloses(currentChunkRange))
+				.filter((r) -> r.length() > currentChunkRange.length())

```

Comment:
Doesn't the r.length have to be greater than the `currentChunkRange` in order for `r.encloses(currentChunkRange)`. Will this ever filter anything out?